### ISSUE TITLE
OrderBy spring-cloud-gcp-starter-data-firestore is not working

### ISSUE THREAD

**Issue by joemersonsouza** [GoogleCloudPlatform/spring-cloud-gcp#1827]
I have a document Resource { id, person: { Name}} and I'm trying to query the collection sorting by person.name but I'm receiving the following error:

"message": "Persistent property does not exist: person.Name",
    "requestId": "5a9ea081-1",
    "trace": "java.lang.IllegalArgumentException: Persistent property does not exist: person.Name\n\tat com.google.cloud.spring.data.firestore.repository.query.PartTreeFirestoreQuery.createFirestoreSortOrders(PartTreeFirestoreQuery.java:190)\n\tSuppressed: The stacktrace has been enhanced by Reactor, refer to additional information below: \nError has been observed at the ...."
    
    The attribute exists and is never null.

**Comment by meltsufin** [comment]
Yes, I think it's a limitation in the current implementation. I would consider it an enhancement request.

**Comment by joemersonsouza** [comment]
Hey fellows, do we have any idea when this issue will be fixed?

**Comment by meltsufin** [comment]
We don't have an ETA, but contributions for this are welcomed.

**Comment by joemersonsouza** [comment]
Cool, I'm going to clone the repo and starting the fix
